math/src/unsigned_integer/element.rs

```diff
@@ -477,15 +477,18 @@ impl<const NUM_LIMBS: usize> ByteConversion for UnsignedInteger<NUM_LIMBS> {
 
     fn from_bytes_be(bytes: &[u8]) -> Result<Self, ByteConversionError> {
         let mut limbs = Vec::with_capacity(NUM_LIMBS);
-        bytes.chunks_exact(8).try_for_each(|chunk| {
-            let limb = u64::from_be_bytes(
-                chunk
-                    .try_into()
-                    .map_err(|_| ByteConversionError::FromBEBytesError)?,
-            );
-            limbs.push(limb);
-            Ok::<_, ByteConversionError>(())
-        })?;
+
+        bytes[0..NUM_LIMBS * 8]
```

Comment:
This could panic if the argument doesn't have enough whole limbs. Please consider using `bytes.get(0..NUM_LIMBS * 8)?` instead, as this allows catching the failure gracefully.

Reply:
Makes sense